src/interface/src/app/plan.service.ts

```diff
@@ -0,0 +1,82 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, catchError, map, Observable, of, take, tap } from 'rxjs';
+
+import { BasePlan, Plan } from './types';
+
+export interface PlanState {
```

Comment:
should this go in plan.types.ts?

Reply:
It can.. but I left it in PlanService because I find it easier to reference it in the same file, since only the service should update PlanState.